File: Golang/Alternating Characters.go
/*
Given a string of characters named s, containing only the characters A and B, this challenge
is to manipulate the string via deletion so there are no repeating characters
*/

func alternatingCharacters(s string) int32 {
    // Variable to count how many matching consecutive characters exist
    var count int32

    // If the character after the one we're looking at the is the same, increase count
    runes := []rune(s)
    for i := 0; i < len(runes)-1; i++ {
        if runes[i] == runes[i+1] {
            count++
        }
    }

    // Return how many matching consecutive characters exist
    return count;
}

File: Python/Alternating Characters.py
# Given a string of characters named s, containing only the characters A and B, this challenge
# is to manipulate the string via deletion so there are no repeating characters

def alternatingCharacters(s):
    # Variables to count how many matching consecutive characters exist, to iterate loop
    count = 0
    i = 0

    # If the character after the one we're looking at is the same, increase count
    while (i < len(s)-1):
        if (s[i] == s[i+1]):
            count += 1
        i += 1

    # Return how many matching consecutive characters exist
    return count
